app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/CustomScriptRow/index.js

```diff
@@ -9,27 +9,113 @@ import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
+import { isFeatureCompatible } from "utils/CompatibilityUtils";
+import FEATURES from "config/constants/sub/features";
 
 const { Text } = Typography;
 
-const CustomScriptRow = ({ rowIndex, pairIndex, isLastIndex, deleteScript, script, scriptIndex, isInputDisabled }) => {
+function getDefaultScript(language, scriptType, isCompatibleWithHMLAttributes) {
+  if (scriptType === GLOBAL_CONSTANTS.SCRIPT_TYPES.URL && isCompatibleWithHMLAttributes) {
+    if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.JS) {
+      return '<!--  Custom attributes to the script can be added here.  -->\n<script type="text/javscript">\n//Everything else will be ignored \n</script>';
+    }
+    if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.CSS) {
+      return '<!--  Custom attributes to the script can be added here.  -->\n<link rel="stylesheet" type="text/css" >\n<!-- Everything else will be ignored  -->\n';
+    }
+  }
+  if (scriptType === GLOBAL_CONSTANTS.SCRIPT_TYPES.CODE) {
+    if (isCompatibleWithHMLAttributes) {
+      if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.JS) {
+        return '<script type="text/javascript">\n\tconsole.log("Hello World");\n</script>';
+      }
+      if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.CSS) {
+        return "<style>\n\tbody {\n\t\t background-color: #fff;\n\t}\n</style>";
+      }
+    } else {
+      if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.JS) {
+        return 'console.log("Hello World");';
+      }
+      if (language === GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.CSS) {
+        return "body {\n\t background-color: #fff;\n }";
+      }
+    }
+  }
+
+  return "";
+}
+
+const createAttributesString = (attributes) => {
+  const attributesString =
+    attributes ??
+    []
+      .map(({ name: attrName, value: attrVal }) => {
+        if (!attrVal) return `${attrName}`;
+        return `${attrName}="${attrVal}"`;
+      })
+      .join(" ");
+  return attributesString;
+};
+
+const CustomScriptRow = ({
+  rowIndex,
+  pairIndex,
+  isLastIndex,
+  deleteScript,
+  script,
+  scriptIndex,
+  isInputDisabled,
+  pair,
+}) => {
   const dispatch = useDispatch();
 
   const [isCodeTypePopupVisible, setIsCodeTypePopupVisible] = useState(false);
+  const [isSourceTypePopupVisible, setIsSourceTypePopupVisible] = useState(false);
   const [codeTypeSelection, setCodeTypeSelection] = useState(GLOBAL_CONSTANTS.SCRIPT_CODE_TYPES.JS);
+  const [sourceTypeSelection, setSourceTypeSelection] = useState(GLOBAL_CONSTANTS.SCRIPT_TYPES.CODE);
   const [isScriptDeletePopupVisible, setIsScriptDeletePopupVisible] = useState(false);
   const [isCodeFormatted, setIsCodeFormatted] = useState(false);
 
-  const scriptEditorBoilerCode = useMemo(
-    () => ({
-      JS: 'console.log("Hello World");',
-      CSS: "body {\n\t background-color: #fff;\n }",
-    }),
-    []
-  );
+  const isCompatibleWithAttributes = isFeatureCompatible(FEATURES.SCRIPT_RULE_HTML_BLOCK);
```

Comment:
Please rename to FEATURES.SCRIPT_RULE.ATTRIBUTES_SUPPORT because we still don't support HTML block. It sounds confusing